File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.java
import java.util.*;
class MyQueue {
   Stack < Integer > input = new Stack < > ();
    Stack < Integer > output = new Stack < > ();
    public MyQueue() {
        
    }
    
    public void push(int x) {

        input.push(x);
    }
    
    public int pop() {
        if(!output.empty()){
            return output.pop();
        }
        else{
            while(!input.empty()){
            int y= input.pop();
            output.push(y);
            }
            return output.pop();
        }
    }
    
    public int peek() {
        if(!output.empty()){
            return output.peek();
        }
        else{
            while(!input.empty()){
            int y= input.pop();
            output.push(y);
            }
        return output.peek();
            
        }
    }
    
    public boolean empty() {
        if(input.empty()&& output.empty()){
            return true;
        }
        else{
            return false;
        }
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */

File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.py
class MyQueue:

    def __init__(self):
        self.l1=[]
        self.l2=[]
        

    def push(self, x: int) -> None:
        self.l1.append(x)

        

    def pop(self) -> int:
        self.peek()
        return self.l2.pop()

        

    def peek(self) -> int:
        if not self.l2:
            while self.l1:
                self.l2.append(self.l1.pop())
        return self.l2[-1]
        

    def empty(self) -> bool:
        return not self.l1 and not self.l2
        


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()